### PR TITLE
[RSDK-8405] [RSDK-8408] Update README to include instructions for FFC-3P and Yolo Detection Network

### PR DESCRIPTION
**Pull request by hexbabe**
Before, this module only supported the `oak-d`, and only included instructions for configuring it.

This PR tries to adds coherent instructions for adding `oak-ffc-3p` and `yolo-detection-network` now that they're almost ready for production.

Can look at the rendered markdown [here](https://github.com/viamrobotics/viam-camera-oak/tree/update-readme)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,97 +1,200 @@
...
 
-Navigate to the **Config** tab of your robot’s page in [the Viam app](https://app.viam.com/).
-Click on the **Components** subtab and click **Create component**.
-Select the `camera` type, then select the `oak` model.
+For camera components, click on the **Components** subtab and click **Create component**.
+Select the `camera` type, then select the `oak-d` or `oak-ffc-3p` model.
 Enter a name for your camera and click **Create**.
 
-On the new component panel, copy and paste the following attribute template into your camera’s **Attributes** box:
+For vision services, click on the **Services** subtab and click **Create service**.
```
Comment (sguequierre) [inline README.md:29]:
```suggestion
To configure the vision service, click on the **Services** subtab and click **Create service**.
```

**Review by sguequierre — CHANGES_REQUESTED** [review]
A great start-- try to follow template [in docs here](https://docs.viam.com/use-cases/create-module/#optional-create-a-readme)! Added some pointers from template.

File: README.md [hunk 2]
```diff
@@ -1,97 +1,200 @@
-# OAK Modular Component
+# OAK Module
 
-This is a [Viam module](https://docs.viam.com/manage/configuration/#modules) for the [OAK](https://shop.luxonis.com/collections/oak-cameras-1) family of cameras. Registered at https://app.viam.com/module/viam/oak.
+This is a [Viam module](https://docs.viam.com/manage/configuration/#modules) for the [OAK](https://shop.luxonis.com/collections/oak-cameras-1) family of cameras. Registered at https://app.viam.com/module/viam/oak. The module supports getting color and stereo depth frames from OAK cameras using the `oak-d` and `oak-ffc-3p` component models, as well as setting up a YOLO detection network on the camera VPU using `yolo-detection-network` vision service model.
```
[the remark this answers is not in the document]
Comment (hexbabe) [reply README.md:3]: The basic agnostic model is being deprecated

File: README.md [hunk 3]
```diff
@@ -1,97 +1,200 @@
...
+| `width_px` | int | Optional | Width in pixels of the images output by this camera. Default: `1280` |
+| `height_px` | int | Optional | Height in pixels of the images output by this camera. Default: `720` |
+| `frame_rate` | int | Optional | The frame rate the camera will capture images at. Default: `30` |
+| `color_order` | str | Optional | The color order of the output frames (used for color sensor type only): "rgb" or "bgr". Default: `rgb` |
+| `interleaved` | bool | Optional | Whether or not output frames should be stored in an interleaved format. Default: `false` |
+
+> [!NOTE]  
+> Higher resolutions may cause out of memory errors. See Luxonis documentation [here](https://docs.luxonis.com/projects/api/en/latest/tutorials/ram_usage/.).
+
+### Configuring the YOLO Detection Network service model
```
[the remark this answers is not in the document]
Comment (hexbabe) [reply README.md]: Done

**Review by randhid — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,73 +1,178 @@
...
+
+The following attributes are available for the `yolo-detection-network` service:
+
+| Name | Type | Inclusion | Description |
+| ---- | ---- | --------- | ----------- |
+| `cam_name` | string | Required | The name of the Viam OAK camera component to set up the service on. |
+| `input_source` | string | Required | The socket name i.e. "cam_a", "cam_b", "cam_c" to get neural network input frames from. Can also be set as "color" to select the primary color sensor on the underlying OAK component. |
+| `num_threads` | int | Optional | How many threads should the DepthAI node use to run the network. |
+| `num_nce_per_thread` | int | Optional | How many Neural Compute Engines should a single thread use for inference. |
+| `yolo_config` | struct | **Required** | A struct mapping of strings to values representing the sub-configuration for the YOLO model. |
```
Comment (randhid) [inline README.md:161]: Are there defaults here?

Comment (hexbabe) [reply README.md:161]: Missed these good catch

Comment (hexbabe) [reply README.md]: Done

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sguequierre) [reply README.md:3]: sounds good-- editing suggestion to reflect

File: README.md [hunk 5]
```diff
@@ -1,73 +1,178 @@
...
-Select the `camera` type, then select the `oak` model.
+[Add `camera` / `oak`,  `camera` / `oak-d` or `camera` / `oak-ffc-3p` to your machine](https://docs.viam.com/configure/#components).
 Enter a name for your camera and click **Create**.
 
-On the new component panel, copy and paste the following attribute template into your camera’s **Attributes** box:
+To configure the vision service, click on the **Services** subtab and click **Create service**.
 
+On the new resource config panel, modify the attributes JSON in the **Attributes** box. If you are confused which attributes to supply, consult the per-model attribute configuration guides below.
+
+## Attribute Configuration Guides
```
Comment (sguequierre) [inline README.md:35]:
```suggestion
## Attribute configuration guides
```

File: README.md [hunk 6]
```diff
@@ -1,73 +1,178 @@
...
+
+### Configuring the `oak-d` component model
+
+#### Example Configuration
```
Comment (sguequierre) [inline README.md:39]:
```suggestion
#### Example configuration
```

File: README.md [hunk 7]
```diff
@@ -1,73 +1,178 @@
...
-Edit these attributes as applicable to your machine. 
-
-> [!NOTE]  
-> For more information, see [Configure a Robot](https://docs.viam.com/manage/configuration/).
-
-## Attributes
+#### Attributes
 
-The following attributes are available for `oak` cameras:
+The following attributes are available for the `oak-d` component:
```
Comment (sguequierre) [inline README.md:60]:
```suggestion
### `oak-d` component model

The following attributes are available for the `oak-d` camera component:
```

File: README.md [hunk 8]
```diff
@@ -1,73 +1,178 @@
...
+| `width_px` | int | Optional | Width in pixels of the images output by this camera. Default: `1280` |
+| `height_px` | int | Optional | Height in pixels of the images output by this camera. Default: `720` |
 | `frame_rate` | int | Optional | The frame rate the camera will capture images at. Default: `30` |
 | `device_info` | string | Optional | Physical device identifier to connect to a specific OAK camera connected to your machine. If not specified, the module will pick the first device it detects. `device_info` can be a MXID, usb port path, or IP address. [See DepthAI documentation for more details](https://docs.luxonis.com/software/depthai/examples/device_information#Device%20information). |
 
 > [!NOTE]  
 > Higher resolutions may cause out of memory errors. See Luxonis documentation [here](https://docs.luxonis.com/projects/api/en/latest/tutorials/ram_usage/.).
 
-### Example Configuration
+### Configuring the `oak-ffc-3p` component model
```
Comment (sguequierre) [inline README.md:73]:
```suggestion
### `oak-ffc-3p` component model
```
To match

File: README.md [hunk 9]
```diff
@@ -1,73 +1,178 @@
...
 
+#### Example Configuration
```
Comment (sguequierre) [inline README.md:75]:
```suggestion
#### Example configuration
```

File: README.md [hunk 10]
```diff
@@ -1,73 +1,178 @@
...
+| `width_px` | int | Optional | Width in pixels of the images output by this camera. Default: `1280` |
+| `height_px` | int | Optional | Height in pixels of the images output by this camera. Default: `720` |
+| `frame_rate` | int | Optional | The frame rate the camera will capture images at. Default: `30` |
+| `color_order` | str | Optional | The color order of the output frames (used for color sensor type only): "rgb" or "bgr". Default: `rgb` |
+| `interleaved` | bool | Optional | Whether or not output frames should be stored in an interleaved format. Default: `false` |
+
+> [!NOTE]  
+> Higher resolutions may cause out of memory errors. See Luxonis documentation [here](https://docs.luxonis.com/projects/api/en/latest/tutorials/ram_usage/.).
+
+### Configuring the `yolo-detection-network` service model
```
Comment (sguequierre) [inline README.md:126]:
```suggestion
### `yolo-detection-network` service model
```
to match

File: README.md [hunk 11]
```diff
@@ -1,73 +1,178 @@
...
+
+#### Example Configuration
```
Comment (sguequierre) [inline README.md:128]:
```suggestion
#### Example configuration
```

File: README.md [hunk 12]
```diff
@@ -1,73 +1,178 @@
...
+| `labels` | list[string] | Required | A list of strings representing each label the YOLO model can output. Order matters here as YOLO models use label index mapping to retrieve string labels. |
+| `confidence_threshold` | float | Required | The minimum confidence level required for a detection to be considered valid. Default `0.5`. |
+| `iou_threshold` | float | Required | The Intersection Over Union (IOU) threshold used for non-max suppression to filter out overlapping bounding boxes. Default `0.5`. |
+| `anchors` | list[float] | Optional | A list of floats representing the anchor boxes used by the YOLO model. The values should be in pairs representing width and height. Default: `[]` |
+| `anchor_masks` | dict | Optional | A dictionary where keys are strings (e.g., "side26", "side13") representing different scales, and values are lists of integers representing the anchor indices used at each scale. Default: `{}` |
+| `coordinate_size` | int | Optional | The number of coordinates used for each bounding box. Typically, this is 4 for (x, y, width, height). Default `4`. |
+
+## Next Steps
```
Comment (sguequierre) [inline README.md:174]:
```suggestion
## Next steps
```

Comment (sguequierre) [reply README.md:174]: Might want to add some actual next steps here besides debugging, like describing how to test the camera in the viam app? quick description would be helpful

**Review by sguequierre — CHANGES_REQUESTED** [review]

**Comment by hexbabe** [comment]
This is looking a lot better. Thanks @sguequierre

File: README.md [hunk 13]
```diff
@@ -1,73 +1,189 @@
...
 > [!NOTE]  
-> Before configuring your camera, you must [create a robot](https://docs.viam.com/manage/fleet/robots/#add-a-new-robot).
+> Before configuring your resource, you must create a new smart machine. See [here](https://docs.viam.com/use-cases/configure/#configure-a-machine) for how to do that and more on the setup and configuration process as a whole.
+
+Navigate to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com/).
 
-Navigate to the **Config** tab of your robot’s page in [the Viam app](https://app.viam.com/).
-Click on the **Components** subtab and click **Create component**.
-Select the `camera` type, then select the `oak` model.
+[Add `camera` / `oak`,  `camera` / `oak-d` or `camera` / `oak-ffc-3p` to your machine](https://docs.viam.com/configure/#components).
```
Comment (sguequierre) [inline README.md:31]:
```suggestion
[Add `camera` / `oak-d` or `camera` / `oak-ffc-3p` to your machine](https://docs.viam.com/configure/#components).
```

File: README.md [hunk 14]
```diff
@@ -1,73 +1,189 @@
...
 Enter a name for your camera and click **Create**.
 
-On the new component panel, copy and paste the following attribute template into your camera’s **Attributes** box:
+To configure the vision service, click on the **Services** subtab and click **Create service**.
```
Comment (sguequierre) [inline README.md:34]:
```suggestion
To configure the vision service, add [``](/configure/#services).
```
This isn't the right instructions. What would be the name of the service? I can't figure out how to add it -- doesn't seem like it's a part of this module?

**Review by sguequierre — CHANGES_REQUESTED** [review]

File: README.md (hunk 13 above)
Comment (sguequierre) [reply README.md:31]: since removing basic model!

File: README.md (hunk 14 above)
Comment (hexbabe) [reply README.md:34]: Unfortunately the new version that actually does all these changes is not released yet.

Comment (hexbabe) [reply README.md:34]: Try it again! I just released a new version

File: README.md (hunk 12 above)
Comment (hexbabe) [reply README.md]: Added how to view outputs in App

File: README.md (hunk 14 above)
Comment (hexbabe) [reply README.md]: Also the suggested change `To configure the vision service, add [``](/configure/#services).` results in this 
Screenshot 2024-08-13 at 4 30 19 PM

**Pull request merged**